src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1296 @@
+            bytes_total: wstat.bytes_total,
+            records_truncated: rstat.records_truncated,
+            duration: start.elapsed(),
+        }),
+    }
+    Ok(())
+}
+
+/// Perform the copy/convert operations. File backed output version
+// Note: Some of dd's functionality depends on whether the output is actually a file. This breaks the Output<Write> abstraction,
+// and should be fixed in the future.
+fn dd_fileout<R: Read>(mut i: Input<R>, mut o: Output<File>) -> Result<(), Box<dyn Error>> {
+    let mut rstat = ReadStat {
+        reads_complete: 0,
+        reads_partial: 0,
+        records_truncated: 0,
+    };
+    let mut wstat = WriteStat {
+        writes_complete: 0,
+        writes_partial: 0,
+        bytes_total: 0,
+    };
+    let start = time::Instant::now();
+    let bsize = calc_bsize(i.ibs, o.obs);
+
+    let prog_tx = {
+        let (tx, rx) = mpsc::channel();
+        thread::spawn(gen_prog_updater(rx, i.xfer_stats));
+        tx
+    };
+
+    while below_count_limit(&i.count, &rstat, &wstat) {
+        // Read/Write
+        let loop_bsize = calc_loop_bsize(&i.count, &rstat, &wstat, i.ibs, bsize);
+        match read_helper(&mut i, loop_bsize)? {
+            (
+                ReadStat {
+                    reads_complete: 0,
+                    reads_partial: 0,
+                    ..
+                },
+                _,
+            ) => break,
+            (rstat_update, buf) => {
+                let wstat_update = o.write_blocks(buf)?;
+
+                rstat += rstat_update;
+                wstat += wstat_update;
+            }
+        };
+        // Update Prog
+        prog_tx.send(ProgUpdate {
+            reads_complete: rstat.reads_complete,
+            reads_partial: rstat.reads_partial,
+            writes_complete: wstat.writes_complete,
+            writes_partial: wstat.writes_partial,
+            bytes_total: wstat.bytes_total,
+            records_truncated: rstat.records_truncated,
+            duration: start.elapsed(),
+        })?;
+    }
+
+    if o.cflags.fsync {
+        o.fsync()?;
+    } else if o.cflags.fdatasync {
+        o.fdatasync()?;
+    }
+
+    match i.xfer_stats {
+        Some(StatusLevel::Noxfer) | Some(StatusLevel::None) => {}
+        _ => print_xfer_stats(&ProgUpdate {
+            reads_complete: rstat.reads_complete,
+            reads_partial: rstat.reads_partial,
+            writes_complete: wstat.writes_complete,
+            writes_partial: wstat.writes_partial,
+            bytes_total: wstat.bytes_total,
+            records_truncated: rstat.records_truncated,
+            duration: start.elapsed(),
+        }),
+    }
+    Ok(())
+}
+
+// The compiler does not like Clippy's suggestion to use &str in place of &String here.
+#[allow(clippy::ptr_arg)]
+fn append_dashes_if_not_present(mut acc: Vec<String>, s: &String) -> Vec<String> {
+    if Some("--") != s.get(0..=1) {
+        acc.push(format!("--{}", s));
+    }
+    acc
+}
```

Comment:
```suggestion
fn append_dashes_if_not_present(mut acc: Vec<String>, mut s: String) -> Vec<String> {
    if !s.starts_with("--") {
        s.insert_str(0, "--");
        acc.push(s);
    }
    acc
}
```
And change the `iter()` in uumain to `into_iter()`. I don't fully understand the context here, but do we want to ignore args that start with `--` already?

Reply:
I like it :-) (I made this change).